openupgrade_scripts/scripts/l10n_fr/18.0.2.1/pre-migration.py

```diff
@@ -0,0 +1,687 @@
+    ("l10n_fr.tax_report_Y3", "l10n_fr_account.tax_report_Y3"),
+    ("l10n_fr.tax_report_Y4", "l10n_fr_account.tax_report_Y4"),
+    ("l10n_fr.tax_report_Y5", "l10n_fr_account.tax_report_Y5"),
+    ("l10n_fr.tax_report_Y6", "l10n_fr_account.tax_report_Y6"),
+    ("l10n_fr.tax_report_Z1", "l10n_fr_account.tax_report_Z1"),
+    ("l10n_fr.tax_report_Z2", "l10n_fr_account.tax_report_Z2"),
+    ("l10n_fr.tax_report_Z3", "l10n_fr_account.tax_report_Z3"),
+    ("l10n_fr.tax_report_Z4", "l10n_fr_account.tax_report_Z4"),
+    ("l10n_fr.tax_report_Z5", "l10n_fr_account.tax_report_Z5"),
+    ("l10n_fr.tax_report_credit", "l10n_fr_account.tax_report_credit"),
+    (
+        "l10n_fr.tax_report_credit_constate",
+        "l10n_fr_account.tax_report_credit_constate",
+    ),
+    ("l10n_fr.tax_report_credit_impute", "l10n_fr_account.tax_report_credit_impute"),
+    ("l10n_fr.tax_report_decompte_tva", "l10n_fr_account.tax_report_decompte_tva"),
+    ("l10n_fr.tax_report_determination", "l10n_fr_account.tax_report_determination"),
+    (
+        "l10n_fr.tax_report_montant_op_realisees",
+        "l10n_fr_account.tax_report_montant_op_realisees",
+    ),
+    (
+        "l10n_fr.tax_report_op_imposables_ht",
+        "l10n_fr_account.tax_report_op_imposables_ht",
+    ),
+    (
+        "l10n_fr.tax_report_op_non_imposables",
+        "l10n_fr_account.tax_report_op_non_imposables",
+    ),
+    ("l10n_fr.tax_report_regularisation", "l10n_fr_account.tax_report_regularisation"),
+    ("l10n_fr.tax_report_reliquat", "l10n_fr_account.tax_report_reliquat"),
+    ("l10n_fr.tax_report_tic_tax", "l10n_fr_account.tax_report_tic_tax"),
+    ("l10n_fr.tax_report_tva_brute", "l10n_fr_account.tax_report_tva_brute"),
+    (
+        "l10n_fr.tax_report_tva_brute_autre",
+        "l10n_fr_account.tax_report_tva_brute_autre",
+    ),
+    ("l10n_fr.tax_report_tva_brute_dom", "l10n_fr_account.tax_report_tva_brute_dom"),
+    (
+        "l10n_fr.tax_report_tva_brute_import",
+        "l10n_fr_account.tax_report_tva_brute_import",
+    ),
+    (
+        "l10n_fr.tax_report_tva_brute_metropo",
+        "l10n_fr_account.tax_report_tva_brute_metropo",
+    ),
+    (
+        "l10n_fr.tax_report_tva_brute_petrolier",
+        "l10n_fr_account.tax_report_tva_brute_petrolier",
+    ),
+    ("l10n_fr.tax_report_tva_deductible", "l10n_fr_account.tax_report_tva_deductible"),
+    (
+        "l10n_fr.display_name_in_footer_param",
+        "l10n_fr_account.display_name_in_footer_param",
+    ),
+    (
+        "l10n_fr.account_reports_fr_statements_menu",
+        "l10n_fr_account.account_reports_fr_statements_menu",
+    ),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_xmlids(env.cr, _xmlids_renames)
+    openupgrade.logged_query(  # just in case `account_fr_fec` was not installed
+        env.cr,
+        """
+        UPDATE ir_module_module
+        SET state='to install'
+        WHERE name = 'l10n_fr_account' AND state='uninstalled'""",
```

Comment:
This is where this can be conflicting, as merging the other module may have an undesired state "to install" while it should be "to upgrade".